### ISSUE TITLE
Package names on multi-lines

### ISSUE THREAD

**Issue by callumbwhyte** [umbraco/Umbraco.Marketplace.Issues#62]
Some packages have really long names which looks a bit ugly in the Marketplace.

<img width="429" alt="Screenshot 2023-09-22 at 11 44 50" src="https://github.com/umbraco/Umbraco.Marketplace.Issues/assets/16511093/a6d0f6e2-b79c-40b0-99ce-490dd3c5b369">

It would be good to clean this up, ideally so it doesn't split over multiple lines.

The component title (screenshot above) has `word-break: break-word;` set so the word will split. 

2 alternative routes that could be taken...

1. Apply `text-overflow: ellipsis;` so the text remains on one line but with an ellipsis if longer than the line width

2. Strip some common prefixes out of the title, e.g. `Umbraco.Community.*` or `Our.Umbraco.*` – in the example above the title would become "SimpleContentApps", which is a little cleaner

In any case a package author can override it with the listing title in `umbraco-marketplace.json`

---
_This item has been added to our backlog AB#33072_

**Comment by deanleigh** [comment]
This is not really the package name I think it pulls from the package ID in umbraco-marketplace.json e.g.` "PackageId": "Umbraco.Community.Templates.UmBootstrap"`
I would prefer a `"PackageTitle": "" in the json file

**Comment by AndyButland** [comment]
Just for reference - we display the following for this field, in order of preference:

- The package title provided in the package owner's JSON field (if available)
- The title provided in the NuGet package (if available)
- The package Id

I like the idea of just removing those two common prefixes, so think it makes sense to do that.

@nathanwoulfe - I'll leave it with you to consider too if there are improvements we can make via the suggested CSS.

**Comment by nathanwoulfe** [comment]
I'd tried the text-overflow approach, but wasn't happy with the result as it meant packages with long names (like Umbraco.Community.SimpleContentApps) would be left the generic part of the name (eg Umbraco.Community.SimpleC..) which doesn't provide anything useful to the user. I'll have a look at something using JS, where I could check the height of the rendered heading, then split the string on a `.` if it is taller than a single deck.

Definitely agree trimming the common prefixes is a good idea (but won't resolve all cases - eg EnterSpeed.Source.UmbracoCms.V8), as well as reminding/encouraging package devs to provide a package title rather than falling through to the Nuget ID.

**Comment by deanleigh** [comment]
I realised yesterday after @AndyButland comments there is a 'Title' (in my template-pack.json see below) as well as the Package ID.
I think there is an equivalent in standard nuget packages as well
```

  <PropertyGroup>
    <PackageType>Template</PackageType>
    <PackageId>Umbraco.Community.Templates.UmBootstrap</PackageId>
    <Title>Umbraco.Community.Templates.UmBootstrap</Title>
```

So I will be testing changing the title to:
`    <Title>UmBootstrap</Title>`

Is there a matching value in umbraco-marketplace.json?

**Comment by AndyButland** [comment]
Yes, it's also called "Title" (details of all fields are [here](https://docs.umbraco.com/umbraco-dxp/marketplace/listing-your-package)).  But there's not need to provide it if you already have a `Title` set in your NuGet package (see my [comment above](https://github.com/umbraco/Umbraco.Marketplace.Issues/issues/62#issuecomment-1732325363)).

It's only necessary if for some reason you wanted a different title listed on nuget.org and the Umbraco Marketplace.

**Comment by deanleigh** [comment]
Thanks for all your help with this @AndyButland 
I had managed to get the same readme working for:

- github
- nuget
- Umbraco marketplace

Which is great

I will try changing the tilte

Also I noticed `"Category": "Themes & Starter Kits"`, is not coming through from umbraco-marketplace.json can that be added in a nuget equivelant?

**Comment by AndyButland** [comment]
> Also I noticed "Category": "Themes & Starter Kits", is not coming through from umbraco-marketplace.json can that be added in a nuget equivelant?

Where is your file hosted please?  It looks like we'd expect it at https://umbootstrap.com/umbraco-marketplace.json - but it doesn't seem to be there.

**Comment by deanleigh** [comment]
Thanks Andy,
you have jogged my memory.

This is a problem as my Project URL is the DotNet Project Template itself and I didn't want users getting additional and unnecessary files on installation e.g. marketplace files

I also do not want to make my Project URL the GitHub repository as the live site is the package.

I'll look at alternative solutions

**Comment by AndyButland** [comment]
We've applied some level of fix to this via the linked PR.  We'll use whatever "Title" is provided either in the `umbraco-marketplace.json` file, or the NuGet package - on the basis that if that's done, the package owner has told us what they want the package title to be.

If we don't have that and we use the package's ID - we remove common prefixes and replace the "." with spaces.  This is now updated in production.  There's an example [here](https://marketplace.umbraco.com/package/umbraco.community.templates.umbootstrap), but more will update as the synchronisation runs.
